source/Weather.cpp

```diff
@@ -85,12 +85,15 @@ void Weather::Step(vector<Visual> &visuals)
 	totalAmount *= currentStrength;
 	visuals.reserve(visuals.size() + totalAmount);
 
+	// If a hazard is system wide then use the flagship as the origin position
+	// for new effects. Damage is still dealt from the weather's origin.
+	Point effectOrigin = hazard->SystemWide() ? flagship : origin;
```

Comment:
```suggestion
	const Point &effectOrigin = hazard->SystemWide() ? flagship : origin;
```
No need to copy it since we aren't modifying it further.